### DISCUSSION TITLE
doubt about Class Critic

### DISCUSSION THREAD

**Discussion by wenjunyang** [hpcaitech/ColossalAI#3031] [Community | Q&A] [answered]
i have a doubt about class Critic in `applications/ChatGPT/chatgpt/nn/critic.py`. in common A2C RL algorithm，Critic is the function of current state $s_i$，Critic should be independent with action $a_i$. but both in `applications/ChatGPT/chatgpt/experience_maker/naive.py`  and `applications/ChatGPT/chatgpt/trainer/ppo.py `, action_mask are passed. that means Critic depend on $s_i$ and $a_i$, is it reasonable?
![image](https://user-images.githubusercontent.com/2840667/223349954-d172672a-0cc2-4fc3-9046-41fae5a71b1c.png)

**Comment by wenjunyang** [comment]
pic source: [here](https://github.com/wangshusen/DRL/blob/master/Slides/4_Policy_3.pdf)，page 57

**Comment by FrankLeeeee** [comment]
Hi @wenjunyang , this `Development | Core` category is used for feature discussion by developers. I will change this to `Community | Q&A` instead. Meanwhile, I have informed our developers in charge of the ChatGPT application and they would respond soon.

**Comment by ht-zhou** [comment, marked as answer]
Hello @wenjunyang , thanks for your feedback. I have consider about it. And find it is actually something wrong here.
We will fix it soon, thanks a lot.

**Reply by ht-zhou** [reply]
@wenjunyang We also invite you to fix it and submit a PR to our repo. Looking forward~

**Reply by wenjunyang** [reply]
ok, i will try
